ecommerce/api.py

```diff
@@ -96,8 +97,10 @@ def get_purchasable_course_run(course_key, user):
             order__user=user,
             course_key=course_run.edx_course_key,
     ).exists():
-        log.warning("Course run %s is already purchased by user %s", course_key, user)
-        raise ValidationError("Course run {} is already purchased".format(course_key))
+        mmtrack = get_mmtrack(user, course_run.course.program)
+        if not has_to_pay_for_exam(mmtrack, course_run.course):
+            log.warning("Course run %s is already purchased by user %s", course_key, user)
+            raise ValidationError("Course run {} is already purchased".format(course_key))
```

Comment:
I this already tests that https://github.com/mitodl/micromasters/blob/8ee1a36d937e769abbde0fc45db50b538add37b3/ecommerce/api_test.py#L104